instance/models/logging_mixin.py

```diff
@@ -0,0 +1,128 @@
+# -*- coding: utf-8 -*-
+#
+# OpenCraft -- tools to aid developing and hosting free software projects
+# Copyright (C) 2015 OpenCraft <xavier@opencraft.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+Instance app models - Logging - Mixins
+"""
+
+# Imports #####################################################################
+
+import logging
+
+from django.db import models
+
+from instance.models.logging_utils import level_to_integer
+
+
+# Constants ###################################################################
+
+# TODO: Don't propagate exceptions & debug data to end users
+PUBLISHED_LOG_LEVEL_SET = ('info', 'warn', 'error', 'exception')
+
+
+# Logging #####################################################################
+
+logger = logging.getLogger(__name__)
+
+
+# Functions ###################################################################
+
+def get_iterator_next_function(iterator, empty_value=None):
+    """
+    Returns a function that return one element from the iterator at a time,
+    and `empty_value` once the iterator is empty
+    """
+    def iterator_next_function():
+        """
+        The function returning the next element
+        """
+        try:
+            return next(iterator)
+        except StopIteration:
+            return empty_value
+
+    return iterator_next_function
+
+
+# Models ######################################################################
+
+class LoggerMixin(models.Model):
+    """
+    Logging facilities - Logs stored on the model & shared with the client via websocket
+    """
+    class Meta:
+        abstract = True
```

Comment:
Again, I think you can just remove the `class Meta` and `models.Model` and just make this mixin a regular class, can't you?